### PR TITLE
Bug 1123567 - Init the state machine with the current storage status and consider the Mount-Fail case r=alive

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1123567

### CODE REVIEW

**Comment by try-server-hook** [comment]
[crh0716 Arthur Chen (crh0716)](https://github.com/crh0716) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b36018bab1b5)

File: apps/system/js/external_storage_monitor.js [hunk 1]
```diff
@@ -242,11 +249,17 @@
...
 
-      this._storage.addEventListener('storage-state-change', this);
-      var msg = '[ExternalStorageMonitor] initEvent(): ' +
-                'monitor external storage name = ' + this._storage.storageName +
-                ', canBeMounted = ' + this._storage.canBeMounted;
-      this.debug(msg);
+      var statusReq = this._storage.storageStatus();
+      statusReq.onsuccess = statusReq.onerror = (evt) => {
+        var curStorageStatus =  evt.target.result;
+        this.enableStorageUnrecognised(curStorageStatus === 'Mount-Fail');
```
Comment (ian-liu) [inline apps/system/js/external_storage_monitor.js]: Nice catch here. :100:

**Comment by ian-liu** [comment]
We might to considerate all of the `Idle(Unmounted)` state. It's possible to be the new `Mound-Fail(Unmounted)` state. If yes, some of the recognised patterns should be revised. 

Such as:
https://github.com/crh0716/gaia/blob/1123567/apps/system/js/external_storage_monitor.js#L179
https://github.com/crh0716/gaia/blob/1123567/apps/system/js/external_storage_monitor.js#L209

I have asked some question from Eden on bugzilla. Let's wait his feedback there. Thanks:)

**Comment by ian-liu** [comment]
Per Eden commented on bugzilla, I think we have to handle these cases.

* actionUnrecognisedAfterFormatted (I see you have handled it.)

https://github.com/crh0716/gaia/blob/1123567/apps/system/js/external_storage_monitor.js#L209
* actionUnrecognisedStorageRemoved
     * Idle(Unmounted) --> NoMedia
     * Enumerate value:
     * 1 --> 0
Should be:
     * Mount-Fail(Unmounted) --> NoMedia
     * Enumerate value:
     * 9 --> 0

**Comment by try-server-hook** [comment]
[crh0716 Arthur Chen (crh0716)](https://github.com/crh0716) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=9b083e01f59d)

**Comment by try-server-hook** [comment]
[crh0716 Arthur Chen (crh0716)](https://github.com/crh0716) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7f862a6430bb)

**Pull request merged**